src/ops/dbcsr_operations.F

```diff
@@ -1479,6 +1479,9 @@ SUBROUTINE dbcsr_copy_into_existing_sym(matrix_b, matrix_a)
          IF (.NOT. found) THEN
             CALL dbcsr_data_clear(dst_data)
          ELSE
+            IF (dbcsr_data_get_size(src_data) .NE. dbcsr_data_get_size(dst_data)) THEN
```

Comment:
I wonder if it's too strict like `.LE` instead of `.NE`? Though, the code in this branch performs `dbcsr_block_partial_copy`.

Reply:
The same check is performed in [dbcsr_copy_into_existing](https://github.com/cp2k/dbcsr/blob/2cdbf1331eb9f0e8265f4ca048cedb125f3e38f9/src/ops/dbcsr_operations.F#L1402) and the regtests are passing. So, I'm fairly sure that this is not an intentional feature.


I stumbled across this by trying to restart an LS_SCF calculation from a density matrix of a different basis set. Instead of complaining about the mismatching block sizes CP2K happly went on to compute garbage :-/

Reply:
Thank you for looking into it. LGTM then!

Reply:
Ready to be merged, I guess.